thirtyinch/src/test/java/net/grandcentrix/thirtyinch/internal/TiActivityPresenterDestroyTest.java

```diff
@@ -0,0 +1,524 @@
+/*
+ * Copyright (C) 2016 grandcentrix GmbH
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.grandcentrix.thirtyinch.internal;
+
+
+import net.grandcentrix.thirtyinch.TiConfiguration;
+import net.grandcentrix.thirtyinch.TiPresenter;
+import net.grandcentrix.thirtyinch.TiView;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import android.os.Bundle;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.HashMap;
+
+import static org.assertj.core.api.Java6Assertions.assertThat;
+import static org.assertj.core.api.Java6Assertions.fail;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+
+public class TiActivityPresenterDestroyTest {
+
+    private class PutInMapAnswer implements Answer<Void> {
+
+        final HashMap<String, String> map = new HashMap<>();
+
+        @Override
+        public Void answer(InvocationOnMock invocation) throws Throwable {
+            final Object[] args = invocation.getArguments();
+            map.put((String) args[0], (String) args[1]);
+            return null;
+        }
+    }
+
+    private class Delegate extends TiActivityDelegate<TestPresenter, TiView> {
```

Comment:
It would be better if we move this into its own class (instead of inner class) and setup all the stuff in the constructor via a builder pattern. 
[These ](https://github.com/grandcentrix/ThirtyInch/pull/35/files#diff-f57717e60ac391f21f2e371fd80e01e6R124) lines looks really crazy and no one knows which parameter is for what :D